File: templates/java-lambda/src/main/java/com/example/Handler.java
package com.example;

import com.amazonaws.services.lambda.runtime.Context;
import com.amazonaws.services.lambda.runtime.RequestHandler;
import java.util.Map;

public class Handler implements RequestHandler<Map<String, Object>, Map<String, Object>> {
  @Override
  public Map<String, Object> handleRequest(Map<String, Object> event, Context context) {
    String name = event.getOrDefault("name", "world").toString();
    return Map.of("message", "Hello, " + name + "!");
  }
}

File: templates/typescript-lambda/src/handler.ts
import type { Handler } from "aws-lambda";

interface Request {
  name?: string;
}

interface Response {
  message: string;
}

export const handler: Handler<Request, Response> = async (event) => {
  const name = event.name ?? "world";
  return {
    message: `Hello, ${name}!`,
  };
};
